Thread-device.md

```diff
@@ -0,0 +1,92 @@
+## How to Set Up a Matter Thread Lighting Device on Nordic nRF52840 Dongle
+
+This guide demonstrates how to build and flash a Matter Thread lighting application firmware onto a Nordic Semiconductor nRF52840 Dongle, 
+thereby transforming the dongle into a Matter Thread device.
+
+Complete the following steps to build and flash the firmware:
+
+### Environment Setup
+- Ubuntu 23.04 OS
+- nRF52840 dongle
+
+### Version tested in this guide
+- Nordic Semiconductor's nRF Connect for Desktop: 4.2.0
+- Nordic Semiconductor's nRF Connect SDK: 2.4.2
+- [Matter](https://github.com/project-chip/connectedhomeip) SDK: 1.1.0.1
+
+### 1. Setting up the Build Environment for nRF Connect SDK and Matter SDK
+
+First, follow [Using native shell for setup](https://github.com/project-chip/connectedhomeip/tree/v1.1.0.1/examples/lighting-app/nrfconnect#using-native-shell-for-setup)
+guide's steps 1 and 2 to install `nRF command-line tools`, `GN meta-build system`, and `nRF Connect SDK`.
+
+Once nRF Desktop is installed, change its permissions, install dependencies, and run it:
+```bash
+chmod +x nrfconnect-4.2.0-x86_64.appimage
+sudo apt install libfuse2
+./nrfconnect-4.2.0-x86_64.appimage
+```
+
+In nRF Connect for Desktop, install the latest `toolchain manager` and `nRF Connect SDK v2.4.2`. 
+After installations, click the down arrow next to the nRF Connect SDK version you installed and select `Open Terminal` or `Open Bash`.
+
+### 2. Installing and Activating Matter SDK:
+Install the Matter SDK with a shallow clone for the Linux platform:
+```bash
+cd ~
+git clone https://github.com/project-chip/connectedhomeip.git --depth=1 --branch=v1.1.0.1
+cd ~/connectedhomeip/
+scripts/checkout_submodules.py --shallow --platform linux
+git submodule update --init
+```
+
+Activate Matter build environment:
+```bash
+cd ~/connectedhomeip/
+
+# Skip all non-core Python requirements:
+sed -i '/^-r requirements/ s/./#&/' ./scripts/setup/requirements.txt
+source scripts/activate.sh
+```
+### 3. Building the Matter Lighting Example Firmware:
+By default, support for DFU (Device Firmware Upgrade) using Matter OTA is enabled.
+To ensure successful firmware building for the dongle, utilize the following command, considering that nRF52840 dongle doesn't have the external flash required for DFU:
+```bash
+cd examples/lighting-app/nrfconnect
+west build -b nrf52840dongle_nrf52840 -- -DCONF_FILE=./prj_no_dfu.conf
+```
+
```

Comment:
I got an error here suggesting that I should upgrade my environment by running `/home/farshid/ncs/v2.4.2/connectedhomeip/scripts/setup/nrfconnect/update_ncs.py --update`. It resolved the error.

```suggestion

Note that the build may fail, suggesting an upgrade of the NCS environment via a given command. Executing the command upgrades the environment and resolves the error.
```